src/v/raft/replicate_batcher.cc

```diff
@@ -78,7 +79,7 @@ replicate_batcher::do_cache_with_backpressure(
   std::optional<model::term_id> expected_term,
   ss::circular_buffer<model::record_batch> batches,
   size_t bytes) {
-    return ss::get_units(_max_batch_size_sem, bytes)
+    return ss::get_units(_max_batch_size_sem, std::min(bytes, _max_batch_size))
```

Comment:
please add a code comment here explaining this `std::min`

Reply:
i mean reference the github issue that noah posted here. in that we have to take out the full 1MB sem units.